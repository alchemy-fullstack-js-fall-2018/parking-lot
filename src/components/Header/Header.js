import React from 'react';
import { Link } from 'react-router-dom';
import ROUTES from '../../routes';
import styles from './Header.css';


export default function Header() {
  return (
    <header>
      <nav className={styles.flexContainer}>
        <Link to={ROUTES.HOME.linkTo()}>Home</Link>
        <Link to={ROUTES.REGISTER_CAR.linkTo()}>Register Car</Link>
        <Link to={ROUTES.SEARCH_CARS.linkTo()}>Search All Cars</Link>
        <Link to={ROUTES.SEARCH_LOT_CARS.linkTo()}>Search Cars in Lot</Link>
      </nav>
      <h1>City Center Parking</h1>
    </header>
  );
}
