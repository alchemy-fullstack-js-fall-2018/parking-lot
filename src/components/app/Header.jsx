import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { config, colors } from '../../site/config';

const StyledHeader = styled.header`
  background-color: ${colors.primary};
  color: ${colors.secondary};
`;

const StyledNav = styled.nav`
  margin: 0px 0px 15px 0px;
`;

const StyledLink = styled(Link)`
  box-sizing: border - box;

  text-decoration: none;
  color: var(--secondary);
  margin: 5px;
  padding: 8px;
  border: 1px solid lightyellow;


  &:hover {
    color: var(--darksecondary);
    border: 1px solid var(--darksecondary);
  }
`;

const Header = () => {

  const links = [
    // { label: 'About', path: '/about' },
    // { label: 'Home', path: '/home' },
    { label: 'Members', path: '/cars/search' },
    { label: 'Lot', path: '/lot/search' },
    { label: 'Register', path: '/cars/new' }
  ];
  const LinkComponents = links.map(link => {
    return (
      <StyledLink key={link.label} to={link.path}>
        {link.label}
      </StyledLink>
    );
  });

  return (
    <StyledHeader>
      <h1>{config.title}</h1>
      <StyledNav>
        {LinkComponents}
      </StyledNav>
    </StyledHeader>
  );
};

export default Header;
