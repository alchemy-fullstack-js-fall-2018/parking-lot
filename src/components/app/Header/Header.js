import React from 'react';
import { Link } from 'react-router-dom';

import styles from './Header.css';

import config from '../../../site/config';

const Header = () => {
  return (
    <header className={styles.header}>
      <h1>{config.title}</h1>
      <nav className={styles.nav}>
        <Link
          to={'/about'}
          className={styles.link}
        >
          About
          </Link>
      </nav>
    </header>
  );
};

export default Header;
