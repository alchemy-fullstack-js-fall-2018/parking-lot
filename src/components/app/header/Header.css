@import '../../../styles/variables.css';

.header {
  background-color: var(--primary);
  color: var(--secondary);
}

.nav {
  margin: 0px 0px 15px 0px;

}

.link {
  box-sizing: border-box;

  text-decoration: none;
  color: var(--secondary);
  margin: 5px;
  padding: 8px;
  border: 1px solid lightyellow;


  &:hover {
    color: var(--darksecondary);
    border: 1px solid var(--darksecondary);
  }
}
