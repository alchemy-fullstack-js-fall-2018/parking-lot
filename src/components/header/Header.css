#header {
  background-color: orange;
  padding: 25px;
  h1 {
    margin-top: 0;
    margin-bottom: 16px;
    text-align: center;
  }

  nav {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
}