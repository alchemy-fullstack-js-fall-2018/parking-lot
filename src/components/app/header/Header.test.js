import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header Component', () => {
  it('should be a header', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
