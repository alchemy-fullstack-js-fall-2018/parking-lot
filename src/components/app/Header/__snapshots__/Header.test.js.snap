// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `
<header
  className="header"
>
  <h1>
    Parking Lot
  </h1>
  <nav
    className="nav"
  >
    <Link
      className="link"
      key="Members"
      replace={false}
      to="/cars/search"
    >
      Members
    </Link>
    <Link
      className="link"
      key="Lot"
      replace={false}
      to="/lot/search"
    >
      Lot
    </Link>
    <Link
      className="link"
      key="Register"
      replace={false}
      to="/cars/new"
    >
      Register
    </Link>
  </nav>
</header>
`;
